src/keboola/http_client/async_client.py

```diff
@@ -0,0 +1,233 @@
+
+        self.limiter = None
+        if max_requests_per_second:
+            one_reqeust_per_second_amount = float(1/int(max_requests_per_second))
+            self.limiter = AsyncLimiter(1, one_reqeust_per_second_amount)
+
+        self.default_headers = default_headers or {}
+        self.backoff_factor = backoff_factor
+
+        self.client = httpx.AsyncClient(timeout=self.timeout, verify=self.verify_ssl, headers=self.default_headers,
+                                        auth=self.auth)
+        
+        if not debug:
+            logging.getLogger("httpx").setLevel(logging.WARNING)
+
+    async def _build_url(self, endpoint_path: Optional[str] = None, is_absolute_path=False) -> str:
+        # build URL Specification
+        url_path = str(endpoint_path).strip() if endpoint_path is not None else ''
+
+        if not url_path:
+            url = self.base_url
+        elif is_absolute_path:
+            url = endpoint_path
+        else:
+            url = urljoin(self.base_url, endpoint_path)
+
+        return url
+
+    async def update_auth_header(self, updated_header: Dict, overwrite: bool = False):
+        """
+        Updates the default auth header by providing new values.
+
+        Args:
+            updated_header: An updated header which will be used to update the current header.
+            overwrite: If `False`, the existing header will be updated with new header. If `True`, the new header will
+                overwrite (replace) the current authentication header.
+        """
+
+        if overwrite is False:
+            self._auth_header.update(updated_header)
+        else:
+            self._auth_header = updated_header
+
+    async def __aenter__(self):
+        await self.client.__aenter__()
+        return self
+
+    async def __aexit__(self, *args):
+        await self.client.__aexit__(*args)
+
+    async def close(self):
+        await self.client.aclose()
+
+    async def _request(
+            self,
+            method: str,
+            endpoint: Optional[str] = None,
+            params: Optional[Dict[str, Any]] = None,
+            headers: Optional[Dict[str, str]] = None,
+            **kwargs
+    ) -> httpx.Response:
+
+
+        is_absolute_path = kwargs.pop('is_absolute_path', False)
+        url = await self._build_url(endpoint, is_absolute_path)
+
+        all_params = {**self.default_params, **(params or {})}
+
+        ignore_auth = kwargs.pop('ignore_auth', False)
+        if ignore_auth:
+            all_headers = {**self.default_headers, **(headers or {})}
+        else:
+            all_headers = {**self._auth_header, **self.default_headers, **(headers or {})}
+            if self.auth:
+                kwargs.update({'auth': self.auth})
+
+        if all_params:
+            kwargs.update({'params': all_params})
+        if all_headers:
+            kwargs.update({'headers': all_headers})
+
+        response = None
+
+        for retry_attempt in range(self.retries + 1):
```

Comment:
I think this is not needed and the retry can be handled by the same logic as in the normal client using httpx.Retry and by starting the client(session) with retry parameter [see example here](https://scrapeops.io/python-web-scraping-playbook/python-httpx-retry-failed-requests/#retry-failed-requests-using-retry-strategy)